### ISSUE TITLE
"Live" model modification

### ISSUE THREAD

**Issue by lucasb-eyer** [VisualComputingInstitute/Beacon8#16]
We initially discussed this, I just put it up so we don't forget about it. We wanted the model to notice when it changes, e.g. a module is added to a container, and invalidate all state optimizers stored in it, so that they are forced to recompile. We shortly thought about  a `dirty` flag or implementing and somehow using `__hash__`.

I have not further thought about it, I even almost forgot it, hence me filing this issue.

**Comment by ikostrikov** [comment]
Yes. It doesn't look difficult to implement, so I suggest to keep it as it is for a while.
